### PR TITLE
Integrate Codecov code coverage reporting

### PR DESCRIPTION
**Pull request by OlenaYefymenko**
This patch adds the `Codecov` tool to automatically track test coverage.
It provides a detailed coverage metric for testing in the project.

Related #18

### CODE REVIEW

**Comment by webknjaz** [comment]
> This patch adds powerful tool to automatically track test coverage across branches and pull requests.

Is this an advertisement by ChatGPT?

**Comment by OlenaYefymenko** [comment]
> > This patch adds powerful tool to automatically track test coverage across branches and pull requests.
> 
> Is this an advertisement by ChatGPT?

To be honest, the idea is mine🙂. I just check the grammar sometimes. I had doubts about the word "powerful". I have confusion when I try to add a description without making it sound like an advertisement.

**Review by webknjaz — CHANGES_REQUESTED** [review]
This won't work in the linters job since it doesn't produce any coverage report files. You should collect coverage from the test runs instead.

**Comment by OlenaYefymenko** [comment]
> This won't work in the linters job since it doesn't produce any coverage report files. You should collect coverage from the test runs instead.

Sorry for being inattentive.

**Comment by webknjaz** [comment]
In the CI log, the upload step doesn't find `coverage.xml`. Is it getting created locally? You could temporarily inject an extra step with something like `ls -l` after running the tests in CI, to see what files are present in the directory in the job.

**Comment by OlenaYefymenko** [comment]
> In the CI log, the upload step doesn't find `coverage.xml`. Is it getting created locally? You could temporarily inject an extra step with something like `ls -l` after running the tests in CI, to see what files are present in the directory in the job.

Initially, there was no local file. Then I ran `python -m coverage run manage.py test`. Now, in the repository, I have the `coverage.xml` file. However, when running tests, I noticed the following issue:

```
  Codecov report uploader 0.7.1
[2024-01-10T13:10:30.878Z] ['info'] => Project root located at: /home/runner/work/django-girls-blog-OlenaEfymenko/django-girls-blog-OlenaEfymenko
[2024-01-10T13:10:30.879Z] ['info'] ->  Token found by environment variables
[2024-01-10T13:10:30.979Z] ['info'] Running coverage xml...
[2024-01-10T13:10:31.065Z] ['info'] Searching for coverage files...
[2024-01-10T13:10:31.088Z] ['info'] Warning: Some files located via search were excluded from upload.
[2024-01-10T13:10:31.088Z] ['info'] If Codecov did not locate your files, please review https://docs.codecov.com/docs/supported-report-formats
[2024-01-10T13:10:31.088Z] ['error'] There was an error running the uploader: No coverage files located, please try use `-f`, or change the project root with `-R`
```

Maybe it's because I don't have tests yet. Is that the reason?

**Comment by webknjaz** [comment]
No, the step you added shows that file in the list. Perhaps, you need to point to it explicitly through `with:`.

You can also save it as an artifact in CI so you'd be able to download it after the workflow run.

**Comment by OlenaYefymenko** [comment]
> No, the step you added shows that file in the list. Perhaps, you need to point to it explicitly through `with:`.
> 
> You can also save it as an artifact in CI so you'd be able to download it after the workflow run.

Finally, It's works :)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,6 +23,12 @@ jobs:
     - name: Run tests
       run: python -m manage test
 
+    - name: Upload coverage reports to Codecov
+      uses: codecov/codecov-action@v3
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+        files: ./coverage.xml
```
Comment (webknjaz) [inline .github/workflows/ci.yml:30]: Let's keep the inputs ordered:
```suggestion
        files: coverage.xml
        token: ${{ secrets.CODECOV_TOKEN }}
```

**Comment by webknjaz** [comment]
Interesting.. I'd expect the auto-discovery mechanism to find this file. This might be a bug in the Codecov action. OTOH, this is fine for now.

**Review by webknjaz — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -23,6 +23,12 @@ jobs:
     - name: Run tests
       run: python -m manage test
 
+    - name: Upload coverage reports to Codecov
+      uses: codecov/codecov-action@v3
```
Comment (webknjaz) [inline .github/workflows/ci.yml:27]: There's v5 of the action now.

**Pull request merged**